### DISCUSSION TITLE
Compiling go backend

### DISCUSSION THREAD

**Discussion by peartobear** [vercel/next.js#13583] [Help]
I have a go backend API that builds using makebuild automation tool. I decided to use Nextjs frontend for that and stumbled upon this: https://vercel.com/docs/runtimes#official-runtimes/go I made a /api directory inside the /pages directory and copy pasted my go backend project there, then I tried to compile my backend via vercel using `npm run dev` but `localhost:3000/api/route` doesn't work. I feel like I'm missing something really basic here. The go backend compiles when built separately using make build.

**Comment by balazsorban44** [comment]
Vercel and Next.js are not the same thing. Although Vercel (the hosting platform) allows go runtime, you probably cannot mix it with Next.js (a JavaScript framework). I think you would need to different deployments for this.

I may be wrong here, but you may be able to put the backend into the same project folder though, as it looks to me that the go runtime expects an `api` directory in the root folder, while Next.js looks for it in `pages/api`. These should still be deployed separately.
